internal/target/dtrack_target.go

```diff
@@ -0,0 +1,112 @@
+package target
+
+import (
+	"context"
+	"encoding/base64"
+	"fmt"
+	"strings"
+
+	"github.com/google/uuid"
+	dtrack "github.com/nscuro/dtrack-client"
+	"github.com/sirupsen/logrus"
+	"github.com/spf13/viper"
+
+	"github.com/ckotzbauer/sbom-operator/internal"
+)
+
+type DependencaTrackTarget struct {
+	baseUrl        string
+	apiKey         string
+	imageToProject map[string]uuid.UUID
+}
+
+func NewDependencaTrackTarget() *DependencaTrackTarget {
+	baseUrl := viper.GetString(internal.ConfigKeyDependencaTrackBaseUrl)
+	apiKey := viper.GetString(internal.ConfigKeyDependencaTrackApiKey)
+	return &DependencaTrackTarget{
+		baseUrl: baseUrl,
+		apiKey:  apiKey,
+	}
+}
+
+func (g *DependencaTrackTarget) ValidateConfig() error {
+	if g.baseUrl == "" {
+		return fmt.Errorf("%s is empty", internal.ConfigKeyDependencaTrackBaseUrl)
+	}
+	if g.apiKey == "" {
+		return fmt.Errorf("%s is empty", internal.ConfigKeyDependencaTrackApiKey)
+	}
+	return nil
+}
+
+func (g *DependencaTrackTarget) Initialize() {
+	client, _ := dtrack.NewClient(g.baseUrl, dtrack.WithAPIKey(g.apiKey))
+	projectsPage, err := client.Project.GetAll(context.TODO(), dtrack.PageOptions{
+		PageNumber: 1,
+		PageSize:   10,
+	})
+	if err != nil {
+		logrus.Errorf("Could not fetch projects: %v", err)
+	}
+
+	g.imageToProject = make(map[string]uuid.UUID)
+	for _, project := range projectsPage.Projects {
```

Comment:
Hi! 👋 I'm the author of `dtrack-client`. Just wanted to point out that this will only get you the first 10 projects in DT (according to your supplied `PageOptions` above). You'd need to step through the pages to get *all* projects. I have an example on how to do this [here](https://pkg.go.dev/github.com/nscuro/dtrack-client#example-package-GetAllFindings).

Reply:
Thx for the hint - forgot to finish this (or add a TODO). Will add the loop.